Source/WebCore/page/Page.cpp

```diff
@@ -756,6 +756,12 @@ void Page::setMainFrameURL(const URL& url)
     m_mainFrameOrigin = SecurityOrigin::create(url);
 }
 
+void Page::setMainFrameURLFragment(String&& fragment)
+{
+    if (!fragment.isEmpty())
```

Comment:
The caller already checks it. There is no need to check it again.
Add an assert instead maybe?

Reply:
> The caller already checks it. There is no need to check it again.

It might be more ergonomic to move the emptiness check in here and not have the caller do it, assuming more callers in the future.

Reply:
Lots of this is left over from when I was trying to figure out why it was crashing, I'll cull it all down.